rascal-lsp/src/main/rascal/lang/rascal/lsp/refactor/rename/Parameters.rsc

```diff
@@ -0,0 +1,93 @@
+@license{
+Copyright (c) 2018-2025, NWO-I CWI and Swat.engineering
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+
+1. Redistributions of source code must retain the above copyright notice,
+this list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright notice,
+this list of conditions and the following disclaimer in the documentation
+and/or other materials provided with the distribution.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+POSSIBILITY OF SUCH DAMAGE.
+}
+@bootstrapParser
+module lang::rascal::lsp::refactor::rename::Parameters
+
+extend framework::Rename;
+import framework::TextEdits;
+extend lang::rascal::lsp::refactor::rename::Common;
+
+import lang::rascal::\syntax::Rascal;
+import analysis::typepal::TModel;
+import lang::rascalcore::check::BasicRascalConfig;
+
+import Relation;
+import Set;
+import util::Maybe;
+
+bool isFormalId(IdRole role) = role in formalRoles;
+
+tuple[type[Tree] as, str desc] asType(IdRole idRole) = <#Name, "formal parameter name"> when isFormalId(idRole);
+
+tuple[set[loc], set[loc]] findOccurrenceFiles(set[Define] _:{<_, _, _, role, _, _>, *_}, list[Tree] focus, set[loc]() getSourceFiles, Tree(loc) getTree, Renamer r) {
+    if (!isFormalId(role)) fail;
```

Comment:
this `fail` will only fail the `if` not the function.

it should be written as:

```
if (isFormalId(role)) {
   return findOccurrenceFilesSymmetric(#Name, "<focus[0]>", getSourceFiles, getTree);
}
fail;
```

or maybe we can write `fail findOcurranceFiles` but I forgot.